PatriotWolf/voting-app
# Feature/base app setup - UI Theme and Base Auth

# What does this PR do?

Setup base code to use MUI for UI and latest Next-Auth for authenticate and authorize

## Type of change


## Changes

- Add dashboard page
- Layout using MUI
- CSS from next init removal
- add api for auth. 
- add middleware to cater auth.

## Feature Flags

-none-

## How to test (for reviewers)

1. In main screen, you click sign in.
2. For this setup, I just put button to skip hassle for login. You can choose admin or user.
3. You will redirect to main page. As admin, you will see link to dashboard and add card. As user, you only logged in without dashboard link and redirected to main if you try to navigate dashboard thru URL.
4. Logout button is displayed on main page if you are logged in.

## Demo


https://github.com/PatriotWolf/voting-app/assets/23714533/37b3cbab-4a1f-41e2-931e-74d6d72a083f
